authentication/native_test.go

```diff
@@ -0,0 +1,164 @@
+package authentication
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go-crypto"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/testsCommon"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNativeAuthClient_NewNativeAuthClient(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil txsigner should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgsNativeAuthClient()
+		args.TxSigner = nil
+		authClient, err := NewNativeAuthClient(args)
+		require.Nil(t, authClient)
+		require.Equal(t, ErrNilTxSigner, err)
+	})
+	t.Run("nil proxy should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgsNativeAuthClient()
+		args.Proxy = nil
+		authClient, err := NewNativeAuthClient(args)
+		require.Nil(t, authClient)
+		require.Equal(t, ErrNilProxy, err)
+	})
+	t.Run("nil private key should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgsNativeAuthClient()
+		args.PrivateKey = nil
+		authClient, err := NewNativeAuthClient(args)
+		require.Nil(t, authClient)
+		require.Equal(t, ErrNilPrivateKey, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgsNativeAuthClient()
+		authClient, err := NewNativeAuthClient(args)
+		require.NotNil(t, authClient)
+		require.Nil(t, err)
+	})
+}
+
+func TestNativeAuthClient_GetAccessToken(t *testing.T) {
+	t.Parallel()
```

Comment:
this t.Run can be removed, thus the t.Parallel which is not needed

Reply:
actually not this one, the one from L61

Reply:
done